### PR TITLE
Update gosec actions.yaml

### PR DESCRIPTION
**Pull request by francis-nijay**
# Description
A few sentences describing the overall goals of the pull request's commits.

# GitHub Issues
List the GitHub issues impacted by this PR:

| GitHub Issue # |
| -------------- |
| https://github.com/dell/csm/issues/1221 |

# Checklist:

- [x] Have you run format,vet & lint checks against your submission?
- [x] Have you made sure that the code compiles?
- [x] Did you run the unit & integration tests successfully?
- [x] Have you maintained at least 90% code coverage?
- [x] Have you commented your code, particularly in hard-to-understand areas
- [x] Have you done corresponding changes to the documentation
- [x] Did you run tests in a real Kubernetes cluster?
- [x] Backward compatibility is not broken

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Please also list any relevant details for your test configuration

### CODE REVIEW

**Review by rodrigobassil — DISMISSED** [review]

File: csireverseproxy/pkg/standaloneproxy/standaloneproxy.go [hunk 1]
```diff
@@ -109,7 +107,7 @@ func newReverseProxy(mgmtServer config.ManagementServer) common.Proxy {
 func newTLSConfig(mgmtServer config.ManagementServer) *tls.Config {
 	// #nosec G402
 	tlsConfig := tls.Config{
-		InsecureSkipVerify: mgmtServer.SkipCertificateValidation,
+		InsecureSkipVerify: mgmtServer.SkipCertificateValidation, // #nosec,G402
```
Comment (anathoodell) [inline csireverseproxy/pkg/standaloneproxy/standaloneproxy.go:110]: can we leave a comment on why we are skipping cert validation here, and i assume telling gosec to skip the check as well.

**Review by anathoodell — DISMISSED** [review]

File: .github/workflows/actions.yaml [hunk 2]
```diff
@@ -20,9 +20,9 @@ jobs:
     runs-on: ubuntu-latest
     steps:
       - name: Checkout the code
-        uses: actions/checkout@v2
+        uses: actions/checkout@v4
       - name: Run gosec to check for security vulnerabilities
-        uses: dell/common-github-actions/gosec-runner@main
+        uses: securego/gosec@master
```
Comment (gallacher) [inline .github/workflows/actions.yaml:25]: Are you ensuring this change is being made across all CSM repositories?

File: .github/workflows/actions.yaml [hunk 3]
```diff
@@ -20,9 +20,9 @@ jobs:
...
         with:
           directories: "./..."
           excludes: "G108,G402"
```
Comment (gallacher) [inline .github/workflows/actions.yaml:28]: Why are we continuing to exclude things? Remove these please and use //#nosec

File: .github/workflows/actions.yaml (hunk 2 above)
Comment (francis-nijay) [reply .github/workflows/actions.yaml:25]: Yes this change is to be in sync with karavi-resiliency

File: csireverseproxy/pkg/standaloneproxy/standaloneproxy.go [hunk 4]
```diff
@@ -109,7 +107,7 @@ func newReverseProxy(mgmtServer config.ManagementServer) common.Proxy {
 func newTLSConfig(mgmtServer config.ManagementServer) *tls.Config {
 	// #nosec G402
 	tlsConfig := tls.Config{
-		InsecureSkipVerify: mgmtServer.SkipCertificateValidation,
+		InsecureSkipVerify: mgmtServer.SkipCertificateValidation, // #nosec,G402 - InsecureSkipVerify needs to be configurable
```
Comment (gallacher) [inline csireverseproxy/pkg/standaloneproxy/standaloneproxy.go:110]: It is configurable.  Need a better explanation here

File: .github/workflows/actions.yaml (hunk 3 above)
Comment (francis-nijay) [reply .github/workflows/actions.yaml]: removed excludes

File: csireverseproxy/pkg/standaloneproxy/standaloneproxy.go (hunk 1 above)
Comment (francis-nijay) [reply csireverseproxy/pkg/standaloneproxy/standaloneproxy.go]: Added comment

File: csireverseproxy/pkg/standaloneproxy/standaloneproxy.go (hunk 4 above)
Comment (francis-nijay) [reply csireverseproxy/pkg/standaloneproxy/standaloneproxy.go]: Updated comment

**Review by anathoodell — APPROVED** [review]

**Review by delldubey — APPROVED** [review]
LGTM

**Pull request merged**